org.eclipse.jdt.ui.tests.refactoring/test cases/org/eclipse/jdt/ui/tests/refactoring/ExtractInterfaceTests.java

```diff
@@ -135,6 +135,40 @@ protected void validatePassingTest(String className, String[] cuNames, String ne
 		assertEqualLines("(interface cu)", getFileContents(getOutputTestFileName(newInterfaceName)), interfaceCu.getSource());
 	}
 
+	protected void validatePassingTest(String className, String[] cuNames, String newInterfaceName, boolean replaceOccurrences, String[] extractedMethodNames, String[][] extractedSignatures, String[] extractedFieldNames, IPackageFragment newPackageFragment) throws Exception {
+		IType clas= getClassFromTestFile(getPackageP(), className);
+
+		ExtractInterfaceProcessor processor= new ExtractInterfaceProcessor(clas, JavaPreferencesSettings.getCodeGenerationSettings(clas.getJavaProject()));
+		Refactoring ref= new ProcessorBasedRefactoring(processor);
+
+		processor.setTypeName(newInterfaceName);
+		assertEquals("interface name should be accepted", RefactoringStatus.OK, processor.checkTypeName(newInterfaceName).getSeverity());
+
+		ICompilationUnit[] cus= new ICompilationUnit[cuNames.length];
+		for (int i= 0; i < cuNames.length; i++) {
+			if (cuNames[i].equals(clas.getCompilationUnit().findPrimaryType().getElementName()))
+				cus[i]= clas.getCompilationUnit();
+			else
+				cus[i]= createCUfromTestFile(clas.getPackageFragment(), cuNames[i]);
+		}
+		processor.setReplace(replaceOccurrences);
+		processor.setAnnotations(fGenerateAnnotations);
+		IMethod[] extractedMethods= getMethods(clas, extractedMethodNames, extractedSignatures);
+	    IField[] extractedFields= getFields(clas, extractedFieldNames);
+		processor.setExtractedMembers(merge(extractedMethods, extractedFields));
+		processor.setPackageFragment(newPackageFragment);
```

Comment:
Maybe it's worthwhile to check if the passed `packageFragment` is null, and if so, don't set the processor's package fragment. Then, the other `validatePassingTest` could just call this one with `newPackageFragment = null`, so there is less repeated code.

I don't know the conventions of this code base well, so maybe this is frowned upon in this code base.

Reply:
currently when the field `fFragment` is null, we will fall back to the subtype's fragment, see https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/355/files#diff-4ff695c2e81ce4025c310ebb45028b14ed249febccde313de659114d6c8130d8R1128.

do you mean 

- we initialize the `fFragment` with the subtype's fragment
- check if the passed `packageFragment` is null. If not, set the `fFragment`
- when executing, we directly use `fFragment`, and there is no need for the current fall back

Reply:
I just meant for this helper method in the tests, instead of having a copy paste of an existing method with an additional parameter, we have the old method (that was copied) call the the new method that you introduced, but pass `null` as the `packageFragment`. Then, only call `processor.setPackageFragment(packageFragment)` if the `packageFragment` that was supplied is non-null. This way there is less code duplication.

Reply:
resolved in https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/355/commits/ffd3b23edae1675249ed33f51128e79a79c9a322 :)